packages/uploadthing/src/internal/handler.ts

```diff
@@ -154,10 +154,18 @@ export const buildRequestHandler = <
     res?: TRuntime extends "pages" ? NextApiResponse : undefined;
   }) => {
     const { router, config } = opts;
-    const upSecret = config?.uploadthingId ?? process.env.UPLOADTHING_SECRET;
-
+    const preferredOrEnvSecret =
+      config?.uploadthingSecret ?? process.env.UPLOADTHING_SECRET;
     const { uploadthingHook, slug, req, res, actionType } = input;
+
     if (!slug) throw new Error("we need a slug");
+
+    if (!preferredOrEnvSecret) {
+      throw new Error(
+        `Please set your preferred secret in ${slug} router's config or set UPLOADTHING_SECRET in your env file`
+      );
+    }
```

Comment:
Not sure if there is a use case for this, but I did some slight rearranging here to get a slightly better error message just in case someone wanted to have separate secret configs for their various routers. If this is not desired I can definitely revert this!